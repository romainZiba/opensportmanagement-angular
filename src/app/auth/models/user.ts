export interface Credentials {
  username: string;
  password: string;
}

export class User {
  constructor(
    public username: string,
    public firstName: string,
    public lastName: string,
    public email: string,
    public phoneNumber: string
  ) {}
}
