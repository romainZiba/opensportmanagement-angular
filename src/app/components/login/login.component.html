<form class="container d-flex flex-column" #form="ngForm" (ngSubmit)="onLogin(username.value, password.value)">
  <div style="text-align:center">
    <img width="300" alt="Angular Logo" src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNTAgMjUwIj4KICAgIDxwYXRoIGZpbGw9IiNERDAwMzEiIGQ9Ik0xMjUgMzBMMzEuOSA2My4ybDE0LjIgMTIzLjFMMTI1IDIzMGw3OC45LTQzLjcgMTQuMi0xMjMuMXoiIC8+CiAgICA8cGF0aCBmaWxsPSIjQzMwMDJGIiBkPSJNMTI1IDMwdjIyLjItLjFWMjMwbDc4LjktNDMuNyAxNC4yLTEyMy4xTDEyNSAzMHoiIC8+CiAgICA8cGF0aCAgZmlsbD0iI0ZGRkZGRiIgZD0iTTEyNSA1Mi4xTDY2LjggMTgyLjZoMjEuN2wxMS43LTI5LjJoNDkuNGwxMS43IDI5LjJIMTgzTDEyNSA1Mi4xem0xNyA4My4zaC0zNGwxNy00MC45IDE3IDQwLjl6IiAvPgogIDwvc3ZnPg==">
  </div>
  <mat-form-field>
    <input #username matInput type="text" i18n-placeholder="The username to log in@@username" placeholder="Username"
           name="username">
  </mat-form-field>
  <mat-form-field>
    <input #password matInput [type]="hide ? 'password' : 'text'"
           i18n-placeholder="The password to log in@@password" placeholder="Enter your password"
           name="password">
    <mat-icon matSuffix (click)="hide = !hide">{{hide ? 'visibility' : 'visibility_off'}}</mat-icon>
  </mat-form-field>
  <button type="submit" i18n="@@login" class="align-self-end" mat-raised-button color="primary">Login</button>
</form>
