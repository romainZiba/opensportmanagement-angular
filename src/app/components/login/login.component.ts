import { Component, EventEmitter, Output } from "@angular/core";
import { Credentials } from "../../model/credentials";

@Component({
  selector: "login-form",
  templateUrl: "./login.component.html",
  styleUrls: ["./login.component.css"]
})
export class LoginComponent {
  hide = true;

  @Output("credentials")
  credentialsEmitter = new EventEmitter<Credentials>();

  onLogin(username: string, password: string): void {
    this.credentialsEmitter.emit(new Credentials(username, password));
  }
}
