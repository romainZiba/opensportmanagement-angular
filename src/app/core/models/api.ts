export interface Api {
  baseUrl: string;
  useMock: boolean;
}
