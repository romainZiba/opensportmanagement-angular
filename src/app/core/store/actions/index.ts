export * from './layout.actions';
export * from './teams.actions';
