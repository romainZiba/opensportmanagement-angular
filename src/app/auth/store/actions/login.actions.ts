import {Action} from '@ngrx/store';
import {Credentials, User} from '../../models/user';

export enum LoginActionsType {
  LOGIN = '[Auth] Login',
  LOGIN_SUCCESS = '[Auth] Login Success',
  LOGIN_FAILURE = '[Auth] Login Failure',
  LOGIN_REDIRECT = '[Auth] Login Redirect',
  LOGOUT = '[Auth] Logout'
}

export class Login implements Action {
  readonly type = LoginActionsType.LOGIN;
  constructor(public payload: Credentials) {}
}

export class LoginSuccess implements Action {
  readonly type = LoginActionsType.LOGIN_SUCCESS;
  constructor(public payload: User) {}
}

export class LoginFailure implements Action {
  readonly type = LoginActionsType.LOGIN_FAILURE;
  constructor(public payload: any) {}
}

export class LoginRedirect implements Action {
  readonly type = LoginActionsType.LOGIN_REDIRECT;
}

export class Logout implements Action {
  readonly type = LoginActionsType.LOGOUT;
}

export type LoginActions =
  | Login
  | LoginSuccess
  | LoginFailure
  | LoginRedirect
  | Logout;
