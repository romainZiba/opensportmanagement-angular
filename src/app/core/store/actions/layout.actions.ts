import { Action } from '@ngrx/store';

export enum SidenavActionsType {
  OPEN_SIDENAV = '[Core] Open Sidenav',
  CLOSE_SIDENAV = '[Core] Close Sidenav',
}

export class OpenSidenav implements Action {
  readonly type = SidenavActionsType.OPEN_SIDENAV;
}

export class CloseSidenav implements Action {
  readonly type = SidenavActionsType.CLOSE_SIDENAV;
}

export type SidenavActions = OpenSidenav | CloseSidenav;
