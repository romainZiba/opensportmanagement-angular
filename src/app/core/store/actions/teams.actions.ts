import { Action } from "@ngrx/store";
import { List } from "immutable";
import { Team } from "../../../model/team";

export enum TeamsActionsType {
  LOAD_TEAMS = "[Core] Load teams",
  LOAD_TEAMS_SUCCESS = "[Core] Load teams success",
  LOAD_TEAMS_FAILED = "[Core] Load teams failed",
  SELECT_TEAM = "[Core] Select team"
}

export class LoadTeams implements Action {
  readonly type = TeamsActionsType.LOAD_TEAMS;
}

export class LoadTeamsSuccess implements Action {
  readonly type = TeamsActionsType.LOAD_TEAMS_SUCCESS;
  constructor(public payload: List<Team>) {}
}

export class LoadTeamsFailed implements Action {
  readonly type = TeamsActionsType.LOAD_TEAMS_FAILED;
  constructor(public payload: any) {}
}

export class SelectTeam implements Action {
  readonly type = TeamsActionsType.SELECT_TEAM;
  constructor(public payload: number) {}
}

export type TeamsActions =
  | LoadTeams
  | LoadTeamsSuccess
  | LoadTeamsFailed
  | SelectTeam;
