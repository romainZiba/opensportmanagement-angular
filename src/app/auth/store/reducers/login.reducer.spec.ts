import {initialState, reducer} from './login.reducer';
import * as loginActions from '../actions';
import {User} from '../../models/user';

describe("LoginReducer", () => {
  const user = new User(
    "username",
    "firstName",
    "lastName",
    "email",
    "phoneNumber"
  );

  describe("undefined action", () => {
    it("should return the default state", () => {
      const action = {} as any;
      const result = reducer(undefined, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe("login action", () => {
    it("should return pending state", () => {
      const credentials = { username: "username", password: "password" };
      const action = new loginActions.Login(credentials);
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe("login success action", () => {
    it("should return logged user", () => {
      const action = new loginActions.LoginSuccess(user);
      const result = reducer(
        { ...initialState, pending: true, error: null },
        action
      );
      expect(result).toMatchSnapshot();
    });
  });

  describe("login failure action", () => {
    it("should return error", () => {
      const error = "error";
      const action = new loginActions.LoginFailure(error);
      const result = reducer(
        { ...initialState, pending: true, error: null },
        action
      );
      expect(result).toMatchSnapshot();
    });
  });

  describe("logout action", () => {
    it("should return initial state", () => {
      const action = new loginActions.Logout();
      const result = reducer({ ...initialState, user }, action);
      expect(result).toMatchSnapshot();
    });
  });
});
