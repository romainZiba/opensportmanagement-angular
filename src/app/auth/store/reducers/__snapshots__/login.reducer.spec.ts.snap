// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginReducer login action should return logged user 1`] = `
Object {
  "error": null,
  "pending": false,
  "user": User {
    "email": "email",
    "firstName": "firstName",
    "lastName": "lastName",
    "phoneNumber": "phoneNumber",
    "username": "username",
  },
}
`;

exports[`LoginReducer login action should return pending state 1`] = `
Object {
  "error": null,
  "pending": true,
  "user": null,
}
`;

exports[`LoginReducer login failure action should return error 1`] = `
Object {
  "error": "error",
  "pending": false,
  "user": null,
}
`;

exports[`LoginReducer login success action should return logged user 1`] = `
Object {
  "error": null,
  "pending": false,
  "user": User {
    "email": "email",
    "firstName": "firstName",
    "lastName": "lastName",
    "phoneNumber": "phoneNumber",
    "username": "username",
  },
}
`;

exports[`LoginReducer logout action should return initial state 1`] = `
Object {
  "error": null,
  "pending": false,
  "user": null,
}
`;

exports[`LoginReducer undefined action should return the default state 1`] = `
Object {
  "error": null,
  "pending": false,
  "user": null,
}
`;

exports[`LoginReducer undefined actions should return the default state 1`] = `
Object {
  "error": null,
  "pending": false,
  "user": null,
}
`;
